src/app/projects/components/project-settings/components/embeddings/embeddings.component.ts

```diff
@@ -0,0 +1,139 @@
+import { Component, Input, OnInit, SimpleChanges } from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { Project } from 'src/app/base/entities/project';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { Attribute } from '../../entities/attribute.type';
+import { DownloadedModel } from '../../entities/downloaded-model.type';
+import { Embedding } from '../../entities/embedding.type';
+import { DataHandlerHelper } from '../../helper/data-handler-helper';
+import { SettingModals } from '../../helper/modal-helper';
+
+@Component({
+  selector: 'kern-embeddings',
+  templateUrl: './embeddings.component.html',
+  styleUrls: ['./embeddings.component.scss'],
+})
+export class EmbeddingsComponent implements OnInit {
+
+  @Input() project: Project;
+  @Input() attributesArrayTextUsableUploaded: Attribute[] = [];
+  @Input() settingModals: SettingModals;
+  @Input() isManaged: boolean;
+  @Input() embeddings: Embedding[] = [];
+  @Input() embeddingHandlesMap: Map<string, any> = new Map<string, any>();
+
+  downloadedModelsList$: any;
+  downloadedModelsQuery$: any;
+  downloadedModels: DownloadedModel[] = [];
+  subscriptions$: Subscription[] = [];
+  dataHandlerHelper: DataHandlerHelper;
+
+  constructor(private projectApolloService: ProjectApolloService, private formBuilder: FormBuilder) {
+    this.dataHandlerHelper = new DataHandlerHelper(this.formBuilder, this.projectApolloService);
+  }
+
+  ngOnInit(): void {
+    this.prepareDownloadedModels();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.attributesArrayTextUsableUploaded = changes.attributesArrayTextUsableUploaded?.currentValue;
+    this.embeddings = changes.embeddings?.currentValue;
+    this.embeddingHandlesMap = changes.embeddingHandlesMap?.currentValue;
+    this.embeddingHandlesMap.forEach((value: any) => {
+      value.forEach((element: any) => {
+        element.isModelDownloaded = this.checkIfModelIsDownloaded(element.configString);
+      });
+    });
+  }
+
+  prepareDownloadedModels() {
+    [this.downloadedModelsQuery$, this.downloadedModelsList$] = this.projectApolloService.getModelProviderInfo();
+    this.subscriptions$.push(
+      this.downloadedModelsList$.subscribe((downloadedModels) => this.downloadedModels = downloadedModels));
```

Comment:
this also needs to recheck the values for embeddingHandlesMap so if a new values is provided by the subscription the values for isModelDownloaded are up to date.

I would create a small function that can be called here as well as in the change method if the ebmedingHandlesMap was changed:

```
checkModelDownloaded(){
    if (!this.embeddingHandelsMap || !this.downloadedModels) return;
    this.embeddingHandlesMap.forEach((value: any) => {
      value.forEach((element: any) => {
        element.isModelDownloaded = this.checkIfModelIsDownloaded(element.configString);
      });
    });

}
```